### ISSUE TITLE
2017.1 EAP Not detecting Relay.QL`...` usage

### ISSUE THREAD

**Issue by hailwood** [jimkyndemeyer/js-graphql-intellij-plugin#60]
Basically as the title says the plugin isn't detecting `Relay.QL` usage and so I'm not getting any help.

I imagine it may be related to this https://github.com/facebook/relay/issues/1537 but I figure this plugin is probably the best place to address this for now and I noticed that importing RelayPublic helped the IDE understand that it is a thing, the graphql query still isn't highlighted?

![image](https://cloud.githubusercontent.com/assets/709773/22807593/a8175342-ef8c-11e6-85e3-ecac6adf627b.png)

Note that individual .graphql files are fine and can query the endpoint, and the code itself works fine in use.

Here's an example code block:

```js
import React from 'react'
import Relay from 'react-relay'
import {Link} from 'react-router'
import gravatar from 'gravatar-url';

class ListItem extends React.Component {

    static propTypes = {
        member: React.PropTypes.object,
        router: React.PropTypes.object,
    }

    render() {
        return (
                <a href={`/view/${this.props.member.id}`}>
                    <Img src={gravatar(this.props.member.email, {size: 500})} alt="User Avatar"/>
                    <p><strong>{this.props.member.name}</strong></p>
                    <p>{this.props.member.email}</p>
                </a>
        )
    }
}

export default Relay.createContainer(ListItem, {
        fragments: {
            member: () => Relay.QL`
                fragment on Member {
                    id
                    name
                    email
                }
            `,
        },
    }
)
```

**Comment by jimkyndemeyer** [comment]
Hi Matthew.

I'd like you to create a new project from scratch, and use Relay.QL in a new js/jsx file. This should help pointpoint whether it's specific to your current project.

Thanks,
Jim.

**Comment by hailwood** [comment]
Hi @jimkyndemeyer 

Here's the new repo for reference https://github.com/hailwood/bare-bones-graphql-error-repo

---

I created a new project from scratch,

ran `yarn add react-relay` to add relay
created a new `test.jsx` file with

```javascript
import Relay from 'react-relay';

Relay.QL`
`
```

Issue still persisted.

---

I created `tester.graphql` to prompt the plugin to create the `graphql.config.json` file.
I filled it out with valid details and ran a query to ensure it was working in .graphql files (it is).

---

I then renamed the file to `test.js` and updated the JS version phpstorm was using to support imports. 
No change.

---

What I did note is that it seems to work fine if I manually inject the language into the template literal.

Just for kicks I also tried it with gql after `yarn add graphql-tag`; and noticed the same behaviour.

```javascript
import gql from 'graphql-tag';

gql`
    viewer {
        
    }
`;
```

What's probably relevant and I apologise for not mentioning it in the initial issue is I am running the latest EAP on ubuntu.

**Comment by jimkyndemeyer** [comment]
I'm unable to reproduce this with stable versions of IntelliJ Idea and WebStorm  (2016.3). This is what I see with your repo:

![image](https://cloud.githubusercontent.com/assets/15837712/22837546/6d9a5b76-efc1-11e6-9f52-a85dba63cb28.png)

Since no one else has reported this as an issue, I'd like you to try with a stable version.

**Comment by hailwood** [comment]
Hey @jimkyndemeyer,

So you're spot on, the stable version picks it up and works perfect...
So I guess the question now is, is there a bug in the EAP, or is there an upcoming change that the plugin will need to be aware of?

I don't mind logging a bug report against the EAP, just not sure what I would say that wouldn't result in them just saying "report it to the plugin author" so would like to do some more debugging with you before I lodge the report if required.

**Comment by hailwood** [comment]
Something that is interesting is comparing the AST from the stable version compared to the EAP.
The EAP is parsing the file quite a bit differently, so I have a feeling that this isn't a bug, but rather is it detecting new elements which is breaking our checks?

![image](https://cloud.githubusercontent.com/assets/709773/22852016/1ccdb454-f098-11e6-8d6c-76f002b8a00c.png)

**Comment by jimkyndemeyer** [comment]
This is a breaking change in their PSI, which renders the EAP and the current plugin version incompatible.

See https://github.com/jimkyndemeyer/js-graphql-intellij-plugin/blob/master/src/main/com/intellij/lang/jsgraphql/ide/injection/JSGraphQLLanguageInjectionUtil.java#L80-L84

I'd prefer not to have multiple versions and branches of the plugin, but that might not be possible with this change.

In any case, I'll work on a solution.

Thanks for catching this.

/Jim

**Comment by jimkyndemeyer** [comment]
Fixed in v1.5.1, pending JetBrains approval.
